flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/serializer/schema/SchemaSerializer.java

```diff
@@ -39,8 +39,7 @@ public class SchemaSerializer extends TypeSerializerSingleton<Schema> {
     /** Sharable instance of the TableIdSerializer. */
     public static final SchemaSerializer INSTANCE = new SchemaSerializer();
 
-    private final ListSerializer<Column> columnsSerializer =
-            new ListSerializer<>(ColumnSerializer.INSTANCE);
+    private ListSerializer<Column> columnsSerializer;
```

Comment:
This could be uninitialized when `SchemaSerializer#copy` was called.

Reply:
Thanks, addressed it and done state compatibility testing using https://github.com/apache/flink-cdc/pull/3447.